pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -134,7 +134,22 @@ func (r *ReconcileComplianceRemediation) reconcileRemediation(instance *compv1al
 	if utils.IsMachineConfig(instance.Spec.Object) {
 		return r.reconcileMcRemediation(instance, logger)
 	}
-	// TODO(jaosorior): Create general flow of applying remediations
+	return r.reconcileGenericRemediation(instance, logger)
+}
+
+// Gets a generic remediation and ensures the object exists in the cluster if the
+// remediation if applicable
+func (r *ReconcileComplianceRemediation) reconcileGenericRemediation(instance *compv1alpha1.ComplianceRemediation, logger logr.Logger) error {
+	if instance.Spec.Apply {
+		obj := instance.Spec.Object
+		found := obj.DeepCopy()
+		err := r.client.Get(context.TODO(), types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}, found)
+		if errors.IsNotFound(err) {
+			err = r.client.Create(context.TODO(), obj)
+		}
+		// TODO(jaosorior): If the object is found, should we update it?
```

Comment:
Update how? I think until we have a mechanism to apply the generic remediations, I don't know what would we do here. The remediation status (Applied/Unapplied/...) is handled in a separate function.

Reply:
We already have a mechanism to apply generic remediations. Generic in the sense that it has to be a kube object.

But what I mean is, let's say that you have a configmap you gotta check for, and the contents don't match the remediation. Right now it only checks if the object (in the example it would be the CM) exists in the cluster, if not, it creates it. But, should we do an `update` operation to overwrite the object? that's the question.

Reply:
That's a good question. Do you reckon we would have many Kinds of remediation objects? if not, we could have a switch that would branch off to comparing the types in a per-type manner. For CMs, we could just compare the key/value pairs. I'm not sure what would be the good fallback, maybe we could just `DeepCompare` a pair of `Spec` attributes or just do nothing.

Of course the downside is that the code would no longer be completely generic..

Reply:
uhm... Maybe we should just leave this as-is for now. And start enhancing once we get more concrete examples (when the need comes). But yeah, so far, it seems that it'll be what we need to do, do comparison depending on the type.